core/src/main/java/org/apache/iceberg/util/TableUtil.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.util;
+
+import java.util.List;
+import org.apache.iceberg.Snapshot;
+import org.apache.iceberg.Table;
+import org.apache.iceberg.TableMetadata;
+import org.apache.iceberg.TableOperations;
+import org.apache.iceberg.relocated.com.google.common.collect.Lists;
+
+public class TableUtil {
```

Comment:
This is okay, but I don't see much utility for these methods other than in the actions, and methods were already added to `BaseAction`. I would probably make these private methods in `BaseAction` instead of adding a utility class.

I think that would be better because we usually try to have util methods either use internals (`TableOperations`, `TableMetadata`) or the public API (`Table`) and not mix the two. Rather than rewrite one of these methods to use `TableMetadata`, I'd just move them.

Reply:
@HeartSaVioR Just to check with you, would it be ok if I moved the methods back into BaseAction?